src/Illuminate/Auth/Access/AuthorizationException.php

```diff
@@ -6,5 +6,18 @@
 
 class AuthorizationException extends Exception
 {
-    //
+    /**
+     * Create a new authorization exception instance.
+     *
+     * @param  string|null  $message
+     * @param  mixed|null  $code
+     * @param  \Exception|null  $previous
+     * @return void
+     */
+    public function __construct($message = '', $code = null, Exception $previous = null)
+    {
+        parent::__construct($message, 0, $previous);
+
+        $this->code = $code;
+    }
```

Comment:
I'm confused. Why do you pass 0 and then set `code` using a property?

Reply:
Exceptions by default only allow numeric code, so we set it to `0` (which is the default as per PHP docs) and then set it manually after. I did see an example somewhere that did exactly this but I can't remember where I saw it.

We could keep it to only allowing numeric codes, but as mentioned `PDOException` does a similar thing where it uses string based code ([which phpunit also recognises](https://github.com/sebastianbergmann/phpunit/issues/244)) and this provides a lot more flexibility.